src/Features/Core/Portable/Completion/Providers/AbstractObjectInitializerCompletionProvider.cs

```diff
@@ -96,5 +88,42 @@ protected virtual bool IsInitializable(ISymbol member, INamedTypeSymbol containi
                 member.MatchesKind(SymbolKind.Field, SymbolKind.Property) &&
                 member.IsAccessibleWithin(containingType);
         }
+
+        private static bool CanSupportObjectInitializer(ISymbol symbol, ISymbol within)
+        {
+            if (symbol is IPropertySymbol propertySymbol)
+            {
+                return propertySymbol.GetMethod != null && !propertySymbol.Type.IsStructType() && HasAccesseblePropertiesOrFields(propertySymbol.Type, within);
+            }
+
+            return false;
+        }
+
+        private static bool HasAccesseblePropertiesOrFields(ITypeSymbol type, ISymbol within)
+        {
+            var types = new HashSet<ITypeSymbol>();
+            return HasAccesseblePropertiesOrFields(type, types);
+
+            bool HasAccesseblePropertiesOrFields(ITypeSymbol typeToCheck, HashSet<ITypeSymbol> alreadyCheckedTypes)
+            {
+                if (alreadyCheckedTypes.Contains(typeToCheck))
+                {
+                    return false;
+                }
+
+                alreadyCheckedTypes.Add(typeToCheck);
+                return typeToCheck.GetBaseTypesAndThis().SelectMany(x => x.GetMembers())
+                    .Where(member => member is IPropertySymbol || member is IFieldSymbol)
+                    .Any(member => member.IsWriteableFieldOrProperty() || 
+                            CanSupportCollectionInitializer(member, within) || 
+                            HasAccesseblePropertiesOrFields(member.GetMemberType(), alreadyCheckedTypes));
+            }
+        }
+
+        private static bool CanSupportCollectionInitializer(ISymbol symbol, ISymbol within)
+        {
+            var type = symbol.GetMemberType();
+            return type != null && type.CanSupportCollectionInitializer(within);
```

Comment:
Optional:

=> symbol.GetMemberType()?.CanSupportCollectionInitializer(within) ?? false;

Reply:
📝 The collection initializer check is unnecessary. All objects supporting collection initialization also support object initializer syntax using empty braces.